hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetadataContext.java

```diff
@@ -345,14 +346,16 @@ else if ( MappedSuperclass.class.isAssignableFrom( mapping.getClass() ) ) {
 					final MappedSuperclassDomainType<Object> jpaType = (MappedSuperclassDomainType<Object>)
 							mappedSuperclassByMappedSuperclassMapping.get( safeMapping );
 
-					applyIdMetadata( safeMapping, jpaType );
+					final Set<String> idProperties = applyIdMetadata( safeMapping, jpaType );
 					applyVersionAttribute( safeMapping, jpaType );
 //					applyNaturalIdAttribute( safeMapping, jpaType );
 
 					for ( Property property : safeMapping.getDeclaredProperties() ) {
-						if ( !safeMapping.isVersioned()
+						if ( !idProperties.contains( property.getName() )
```

Comment:
I would do the same thing we do for entities here, better to have consistency (taken from just a few lines above):

```suggestion
						if ( property.getValue() == safeMapping.getIdentifierMapper() ) {
							// property represents special handling for id-class mappings but we have already
							// accounted for the embedded property mappings in #applyIdMetadata &&
							// #buildIdClassAttributes
							continue;
						}
						if ( safeMapping.isVersioned() && property == safeMapping.getVersion() ) {
							// skip the version property, it was already handled previously.
							continue;
						}
						buildAttribute( property, jpaMapping );
```
You can also extract this to a shared method, I believe they should behave the same. @cigaly please check if the equality condition is enough to work with `@IdClass` attributes, if not it would be great if you could also test if it works for `@Entity` mappings and see how it can be handled.

Reply:
I'll check that, but I am not sure that it will work. First test failed.

Reply:
I'm curious why `property.getValue() == safeMapping.getIdentifierMapper()` didn't work, care to elaborate on that? If it works for entities, it would ideally also work for mapped-superclasses. Have you verified if a similar problem also exists on entities or not?

Reply:
In this test case `property.getValue()` returns `BasicValue` object, while `safeMapping.getIdentifierMapper()` returns `Component`, so there is no equality.
We can get list of component properties with `org.hibernate.mapping.Component#getProperties`, but it can contain different instance than `property`. That is why I am comparing property names.

Reply:
> In this test case property.getValue() returns BasicValue object

Gotcha, but looking at the check for entities it seems like we _expect_ the value of identifier properties contained in an id-class to return the `Component` used as mapper, not a `BasicValue`. Could you verify if that's really the case for entities? And if so, why doesn't it work for mapped-superclasses?

Reply:
One possibility is that bug is in the `MappedSuperclass` handling. In this test case `getDeclaredProperties` is returning `year`, `month`, and`value`, while for (identical) `PersistentClass` `getDeclaredProperties` is returning synthetic property `_identiferMapper`, and `value`.

Reply:
@mbladel Now the question is why `MappedSuperclass` should have `_identifierMapper` synthetic property or not? Now it does not.

Reply:
I don't think it should, that is created during a phase of binding that is only done for entities, it might be overly complicated to try to do this for mapped-supers as well. I was just hoping to find a better way to check, without needing to check property names.

I guess what you did is fine for now, just left a suggestion with a slightly different implementation.

Reply:
Thanks, I did noticed `getPropertyNames` method.